cmake/coverage.cmake

```diff
@@ -24,7 +24,8 @@ if(ENABLE_COVERAGE)
         ${CMAKE_SOURCE_DIR}/src/lib_gui/tests -e ${CMAKE_SOURCE_DIR}/src/lib_utility/tests -e
         ${CMAKE_SOURCE_DIR}/src/messaging/tests -e ${CMAKE_SOURCE_DIR}/src/scheduling/tests -e
         ${CMAKE_SOURCE_DIR}/src/test --html-nested=${CMAKE_BINARY_DIR}/coverage/index.html --gcov-delete -j
-        ${CORES_COUNT} ${CMAKE_BINARY_DIR}
+        ${CORES_COUNT} --gcov-executable gcov-12 --exclude-unreachable-branches --exclude-throw-branches
+        ${CMAKE_BINARY_DIR}
```

Comment:
Could you explain?

Reply:
--gcov-executable gcov-12 to force it to use that version to avoid issue while working with docker container.
--exclude-unreachable-branches --exclude-throw-branches to enhance coverage report